src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/documents/DefendantResponseReceiptServiceIT.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.cmc.claimstore.documents;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.cmc.claimstore.BaseMockSpringTest;
+import uk.gov.hmcts.cmc.claimstore.helper.HTMLTemplateProcessor;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.response.DefenceType;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleResponse;
+import uk.gov.hmcts.cmc.email.EmailService;
+import uk.gov.hmcts.reform.pdf.service.client.PDFServiceClient;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.cmc.claimstore.helper.FileUtils.readFile;
+import static uk.gov.hmcts.cmc.claimstore.helper.FileUtils.writeFile;
+import static wiremock.org.custommonkey.xmlunit.XMLAssert.assertXMLEqual;
+
+@RunWith(SpringRunner.class)
+public class DefendantResponseReceiptServiceIT extends BaseMockSpringTest {
+
+    private static final String TIMESTAMP_ATTRIBUTE_VALUE = "timestamp";
+    private static final String DUMMY_TIMESTAMP = "TIMESTAMP";
+
+    @Autowired
+    private DefendantResponseReceiptService defendantResponseReceiptService;
+
+    @Autowired
+    protected HTMLTemplateProcessor htmlTemplateProcessor;
+
+    @MockBean
+    protected EmailService emailService;
+
+    @Test
+    public void shouldGenerateDefenceResponseForClaimWithFullDefenceAlreadyPaid() throws Exception {
+
+        provideLocalPdfService();
+
+        Claim claim = SampleClaim.builder()
+            .withResponse(
+                SampleResponse.FullDefence
+                    .builder()
+                    .withDefenceType(DefenceType.ALREADY_PAID)
+                    .withMediation(null)
+                    .build()
+            )
+            .withRespondedAt(LocalDateTime.now())
+            .build();
+
+        byte[] actualHtmlBytes = defendantResponseReceiptService.createHtml(claim);
+
+        String actualHtml = replaceTimestampsWithFixedValues(new String(actualHtmlBytes));
+
+        // Useful for debugging test ie diff comparison of actual with expected HTML file
+        writeFile("build/tmp/actual.html", actualHtml);
```

Comment:
Not sure this should be kept in the code; but if it is, it can be refactored to be guaranteed platform-independent.

Change the signature to `writeFile(String actualHtml, String firstPath, String... morePaths)`
and build the path with `Paths.get(firstPath, morePaths)` ([javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Paths.html#get(java.lang.String,java.lang.String...)))
and call the method as `writeFile(actualHtml, "build", "tmp", "actual.html")`

You can also read the contents of a file more simply like this: `Files.readString(path)` ([javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#readString(java.nio.file.Path)))

Reply:
Simplified IO in next commit